### PR TITLE
New option to the JSON function to serialize unwrapped arrays

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
Common request: ability of the JSON function to serialize [1,2,3] as `[1,2,3]` instead of `[{"Value":1},{"Value":2},{"Value":3}]`

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs [hunk 1]
```diff
@@ -363,6 +367,25 @@ public void TexlFunctionTypeSemanticsRenameColumns_Negative(string expression)
...
+
+            var r = RecordType.Empty()
+                .Add(new NamedFormulaType("x", FormulaType.String))
+                .Add(new NamedFormulaType("y", FormulaType.UntypedObject));
+
+            var check = engine.Check(parse, r);
+            Assert.True(check.IsSuccess);
+
+            Assert.IsType<StringType>(check.ReturnType);
+        }
```
Comment (Shpakh-MSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs:387]: What is this testcase for?

File: src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs [hunk 2]
```diff
@@ -254,6 +258,17 @@ public void Visit(TableValue tableValue)
                 {
                     _writer.WriteStartArray();
 
+                    var isSingleColumnValueTable = false;
+                    if (_unwrapValueTables)
+                    {
+                        var fieldTypes = tableValue.Type.GetFieldTypes();
+                        var firstField = fieldTypes.FirstOrDefault();
+                        if (firstField != null && !fieldTypes.Skip(1).Any() && firstField.Name.Value == "Value")
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs:266]: >"Value" [](http://example.com/codeflow?start=104&length=7)

We don't have any existing constant for it?

File: src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs (hunk 1 above)
Comment (CarlosFigueiraMSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs:387]: This should not be here, some local tests. Will revert prior to getting this PR out of draft

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/JSON.txt [hunk 3]
```diff
@@ -123,3 +123,25 @@ Error({Kind:ErrorKind.Div0})
 // Error records
 >> JSON(Filter([-2,-1,0,1,2], 1/Value>0))
 Error({Kind:ErrorKind.Div0})
+
+// Unwrapped tables
+>> JSON([1, 2, 3], JSONFormat.UnwrapValueTables)
+"[1,2,3]"
```
Comment (Shpakh-MSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/JSON.txt:129]: You may want to add roundtrip test

Table(ParseJson(JSON([1,2,3])))

**Review by Shpakh-MSFT — DISMISSED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs (hunk 2 above)
Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs:266]: Updating with constant

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/JSON.txt (hunk 3 above)
Comment (CarlosFigueiraMSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/JSON.txt:115]: The result will be a table of untyped values - which are currently not supported in the JSON function.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Json.cs [hunk 4]
```diff
@@ -120,6 +122,34 @@ public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[
...
+                        switch (option)
+                        {
+                            case _ignoreBinaryDataEnumValue:
+                                ignoreBinaryData = true;
+                                break;
+                            case _ignoreUnsupportedTypesEnumValue:
+                                ignoreUnsupportedTypes = true;
+                                break;
+                            case _includeBinaryDataEnumValue:
+                                includeBinaryData = true;
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Json.cs:136]: This is getting overwritten below at 154.  Why do we need to set this? Same for other flags.

Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Json.cs:136]: Good catch, the later code should be removed.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**